tests/test_legacy.py

```diff
@@ -298,9 +298,39 @@
 
 @pytest.mark.parametrize("json_body, clauses", tests)
 def test_json_to_snuba(json_body: Mapping[str, Any], clauses: Sequence[str]) -> None:
-    expected = "\n".join(clauses)
-    query = json_to_snql(json_body, "sessions")
-    assert query.print() == expected
+    request = json_to_snql(json_body, "sessions")
+    request.validate()
+    assert request.app_id == "legacy"
+    assert request.flags is not None
+    assert request.flags.legacy is True
+
+    query_clauses = [c for c in clauses if not c.startswith("--")]
+    query_expected = "\n".join(query_clauses)
+    assert request.query.print() == query_expected
+
+    # Rather than rewrite all these tests to have the new format, transform the
+    # old format to the new one. This captures all the flags that were using
+    # the -- print format, and asserts they are in the request flags
```

Comment:
let's make sure we have a ticket to convert the tests rather than doing this processing